api/catalog/filter.go

```diff
@@ -6,15 +6,57 @@ import (
 	"net/http"
 	"time"
 
+	"fmt"
+
 	"github.com/rancher/norman/api/access"
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/types"
 	"github.com/rancher/types/client/management/v3"
+	"path"
+	"strings"
+	"github.com/sirupsen/logrus"
 )
 
 func TemplateFormatter(apiContext *types.APIContext, resource *types.RawResource) {
 	delete(resource.Values, "icon")
+	resource.Values["versionLinks"] = extractVersionLinks(apiContext, resource)
+	delete(resource.Values, "version")
 	resource.Links["icon"] = apiContext.URLBuilder.Link("icon", resource)
+	catalogName := strings.Split(resource.ID, "-")[0]
+	// idk why current returns a url start with http:/ instead of http://
+	// but this link is hacky
+	resource.Links["catalog"] = strings.Replace(path.Join(path.Dir(apiContext.URLBuilder.Current()), "catalogs", catalogName), "http:/", "http://", 1)
```

Comment:
My eyes.... my poor eyes are bleeding.  Add a method to URLBuilder for `ResourceLinkByID(schema *Schema, id string)`